challenges/cloud/sre.md

```diff
@@ -0,0 +1,376 @@
+# Summary
+
+Monitor, deploy, and extend a prototype service that provides an API for Kubernetes resources
+
+# Rationale
+
+This exercise has two goals:
+
+* It helps us to understand what to expect from you as a Site Reliability Engineer,
+  how you reason about a production service, how you reason about system design, your
+  opinions on automation & tooling, and how you communicate when trying to solve problems.
+* It helps you get a feel for what it would be like to work at Teleport, as this
+  exercise aims to simulate our day-as-usual and expose you to the type of work
+  we're doing here.
+
+We believe this technique is not only better, but also is more fun compared to
+whiteboard/quiz interviews, so common in the industry. It's not without the
+downsides - it could take longer than traditional interviews.
+
+[Some of the best teams use coding challenges.](https://sockpuppet.org/blog/2015/03/06/the-hiring-post/)
+
+We appreciate your time and are looking forward to hacking on this project together.
+
+# Requirements
+
+This challenge covers 5 engineering levels at Teleport. Level 6 is only for internal promotions. Check the
+[engineering levels document](https://raw.githubusercontent.com/gravitational/careers/main/levels.pdf)
+for more details.
+
+* Your submission will need to meet the requirements of the level you are applying for.
+* Split the submission into 2-3 pull requests for us to review. We will review
+  every pull request and provide our feedback.
+* We are going to inspect the repository, test it, and get back to you.
+
+## Server
+
+An incomplete implementation of a Go server is located [here](sre-server.go).
+You will extend this implementation by adding features described below depending
+on the level you are applying to. Start by creating a new Github repository. Then
+let the interview panel know the repository's location by pasting a link in your
+interview Slack channel. Invite interview panel participants as contributors to
+the new repository if you prefer to keep your submission private.
+
+## Design Doc
+
+Start with a brief design document that covers the edge cases and design approach. At
+Teleport, we prefer Markdown for
+[our designs](https://github.com/gravitational/teleport/blob/master/rfd/0000-rfds.md).
+
+Be sure to cover the following in your design:
+
+* Developer workflow
+* Level 3+: Build, Release
+* Level 4+: State management and Delivery
+* Level 5+: Reconciliation, Conflicts, and Automation
+
+Open a pull request with your design and share a link with the reviewers via Slack.
+After the document is approved by at least 2 reviews, move on to the implementation.
+
+A few notes about the design document:
+
+* We expect the design document to be completed roughly within the first week.
+  This is to ensure you have enough time to work on the implementation.
+* Avoid writing an overly detailed design document. Two to three pages is
+  sufficient.
```

Comment:
```suggestion
* Avoid writing an overly detailed design document. 500-1500 words is
  sufficient.
```

Markdown doesn't neatly map to the pages, making 2-3 pages rather ambiguous.  I'm spitballing words (~500 per page) as an alternative length metric, but I'm open to other ideas.

Reply:
Great suggestion Walt.